web/add/key/index.php

```diff
@@ -0,0 +1,120 @@
+<?php
+error_reporting(NULL);
+$TAB = 'USER';
+
+//check for valid format ssh key. Doesn't check it is working! 
+//https://gist.github.com/jupeter/3248095
+function validateKey($value) 
+{
+  $key_parts = explode(' ', $value, 3);
+
+  if (count($key_parts) < 2) {
+    return false;
+  }
+
+  if (count($key_parts) > 3) {
+    return false;
+  }
+
+  $algorithm = $key_parts[0];
+  $key = $key_parts[1];
+
+  if (!in_array($algorithm, array('ssh-rsa', 'ssh-dss'))) {
+    return false;
+  }
+  $key_base64_decoded = base64_decode($key, true);
+
+  if ($key_base64_decoded == FALSE) {
+    return false;
+  }
+
+  $check = base64_decode(substr($key,0,16));
+  $check = preg_replace("/[^\w\-]/","", $check); 
+
+  if((string) $check !== (string) $algorithm) {
+    return false;
+  }
+  return true;
+}
+
+// Main include
+include($_SERVER['DOCUMENT_ROOT']."/inc/main.php");
+
+// Check POST request
+if (!empty($_POST['ok'])) {
+
+    // Check token
+    if ((!isset($_POST['token'])) || ($_SESSION['token'] != $_POST['token'])) {
+        header('location: /login/');
+        exit();
+    }
+
+	if (empty($_POST['v_key'])) $errors[] = __('ssh key');
+    
+        if (!empty($errors[0])) {
+        foreach ($errors as $i => $error) {
+            if ( $i == 0 ) {
+                $error_msg = $error;
+            } else {
+                $error_msg = $error_msg.", ".$error;
+            }
+        }
+        $_SESSION['error_msg'] = __('Field "%s" can not be blank.',$error_msg);
+    }
+    
+    if(!$_SESSION['error_msg']){
+  	//key if key already exisits 
+  	exec (HESTIA_CMD . "v-list-user-ssh-key ".$user." json", $output, $return_var);
+  	$data = json_decode(implode('', $output), true);
+  	$keylist = array();
+  	foreach($data as $key => $value){
+	  	$idlist[] = trim($data[$key]['ID']);
+	  	$keylist[] = trim($data[$key]['KEY']);
+	  	
+  	}
+  	
+  	if(!validateKey($_POST['v_key'])){
+	  	$_SESSION['error_msg']  = __('SSH KEY is invalid');
+  	}
```

Comment:
If the key is invalid is there a reason to process the file further?

Reply:
No,  will change it to switch statement. Then break can be used. 
2nd option would be put every thing in a proper class structure but than a better idea is to rewrite the whole code base.